third_party/move/move-vm/integration-tests/src/tests/runtime_reentrancy_check_tests.rs

```diff
@@ -61,73 +63,24 @@ fn make_dispatch_d_native() -> NativeFunction {
     })
 }
 
-fn compile_and_publish(storage: &mut InMemoryStorage, code: String) {
-    let mut units = compile_units(&code).unwrap();
-    let m = as_module(units.pop().unwrap());
-    let mut blob = vec![];
-    m.serialize(&mut blob).unwrap();
-    storage.publish_or_overwrite_module(m.self_id(), blob);
+fn set_up_storage(modules: Vec<String>) -> (InMemoryStorage, LocalModuleBytesStorage) {
+    let mut resource_storage = InMemoryStorage::new();
+    let mut module_bytes_storage = LocalModuleBytesStorage::empty();
+
+    for module in modules {
+        let mut units = compile_units(&module).unwrap();
+        let m = as_module(units.pop().unwrap());
+        let mut blob = vec![];
+        m.serialize(&mut blob).unwrap();
+
+        resource_storage.publish_or_overwrite_module(m.self_id(), blob.clone());
+        module_bytes_storage.add_module_bytes(m.self_addr(), m.self_name(), blob.into());
```

Comment:
can we re-use resource storage as a backend instead of LocalModuleBytesStorage or have asserts (afaiu module bytes should be used by v2?)

Reply:
Hm, I like the separation that one is for data, one is for code. Although your point is that at low level, we need state view essentially...

Reply:
Yes, let's do that (although it is no longer resource storage, but any storage), I would delegate this to the next PR because I touch it for the things stacked on top everywhere in tests, is it fine?

Reply:
sounds good.